agent/agent_util.go

```diff
@@ -45,4 +46,32 @@ func getUsername() (string, error) {
 	} else {
 		return userInfo.Username, nil
 	}
-}
\ No newline at end of file
+}
+
+func getHostIPAddrs() ([]string, error) {
+    ifaces, err := net.Interfaces()
+    if err != nil {
+        return nil, err
+    }
+    var ipAddrs []string
+    for _, i := range ifaces {
+        addrs, err := i.Addrs()
+        if err != nil {
+            return nil, err
+        }
+        for _, addr := range addrs {
+            var ip net.IP
+            switch v := addr.(type) {
+            case *net.IPNet:
+                    ip = v.IP
+            case *net.IPAddr:
+                    ip = v.IP
+            }
+            ipv4 := ip.To4()
+            if ipv4 != nil && !ip.IsLoopback() {
```

Comment:
What was the reason for not filtering out the unspecified address 0.0.0.0?

Reply:
I don't think 0.0.0.0 is a valid address for an interface to bind to, so it shouldn't appear here.